scripts/build_docs.py

```diff
@@ -0,0 +1,130 @@
+from typing import Dict
+import os
+from pathlib import Path
+from subprocess import check_output
+import argparse
+
+from ruamel.yaml import YAML
+
+
+exclude_files = [".DS_Store", "__init__.py", "__init__.pyc", "README.md", "version.py", "run.py"]
```

Comment:
Is there documentation in any `__init__.py` files that we want to keep?  If so, where should it go?

Reply:
Also, why exclude `run.py`?

Reply:
We shouldn't have docs in `__init__` in my opinion, to make discoverability easier. `run.py` is excluded because no user will actually use this, as it is installed as an entrypoint in `setup.py`, meaning the entrypoint is `allennlp`.

Reply:
There are a few cases where we have non-trivial logic in an `__init__` file.  It'd be nice if those showed up in the docs somehow.  Not really sure how to accomplish this.  I'm fine with moving whatever docs are in an existing (trivial) `__init__` file to a better location.

On `run.py`, it might be nice to include it with some better documentation around "if you want to write your own entrypoint to allennlp, instead of using ours, you might take this script as a starting place."

Reply:
Are there other cases apart from the `PytorchXtoXWrappers` in `modules/etc`? Those probably shouldn't be in the docs.

We don't want people to be writing "entrypoints" for allennlp (we wan them to just write scripts, using allennlp as a library) and anyway, `run.py` is not the place to start for someone writing a script.

Reply:
I just checked the others I was thinking of (initializers, learning rate schedulers, regularizers), and the logic I was thinking of is not actually inside an `__init__` file, so that was a false alarm.

By "entrypoint" I just meant "script", but I just checked what's in `run.py`, and you're right, that's not a good example, and it doesn't need to be in the docs.